src/common/common/UrlUnicityChecker.py

```diff
@@ -0,0 +1,32 @@
+# -*- coding: utf-8 -*-
+
+import hashlib
+import os
+import pickle
+
+
+class UrlUnicityChecker(object):
+
+    def __init__(self, persist_file):
+        self.persist_file = persist_file
+        self.hashed_urls_set = set()
+        self._load()
+
+    def is_unique(self, url):
+        prev_len = len(self.hashed_urls_set)
```

Comment:
Perhaps the name of the method should make explicit that it has a side effect.
something like "is_unique_and_add" + docstring like "if unique: add url to already seen urls and return true, else return false"